alacritty_terminal/src/term/mod.rs

```diff
@@ -860,14 +878,18 @@ impl<T: EventListener> Handler for Term<T> {
             }
 
             // Write full width glyph to current cursor cell.
-            self.write_at_cursor(c).flags.insert(Flags::WIDE_CHAR);
+            self.grid.cursor.template.flags.insert(Flags::WIDE_CHAR);
+            self.write_at_cursor(c);
+            self.grid.cursor.template.flags.remove(Flags::WIDE_CHAR);
 
             // Write spacer to cell following the wide glyph.
             self.grid.cursor.point.column += 1;
-            self.write_at_cursor(' ').flags.insert(Flags::WIDE_CHAR_SPACER);
+            self.grid.cursor.template.flags.insert(Flags::WIDE_CHAR_SPACER);
+            self.write_at_cursor(' ');
+            self.grid.cursor.template.flags.remove(Flags::WIDE_CHAR_SPACER);
```

Comment:
I don't completely understand why you moved this to the template, but I'm completely fine assuming it was done for a good reason. Let me know if you care to explain.

Reply:
The `write_at_cursor` function previously returned a mutable reference. That made borrowing semantics complicated since branching while returning a mutable reference prevents you from reasigning that variable since borrowck assumes that is incorrect (which it is not).

So I found this alternative solution which allows us to do the same amount of work (except for one more branch) in our hottest path.